src/smc-webapp/projects.cjsx

```diff
@@ -981,8 +998,27 @@ load_recent_projects = =>
     if redux.getTable('projects')._table.get().size == 0
         await load_all_projects()
 
-#if not COCALC_MINIMAL
-load_recent_projects()
+
+if not COCALC_MINIMAL
+    load_recent_projects()
+
+
+_project_tables = {}
+_previous_project_id = undefined
+
+load_single_project = (project_id) =>
```

Comment:
Document what this function does.  I don't know.  It's name is "load_single_project" and in your notes you talk about it "merging what's in the store".  But the code you have here is just deleting the Table and making a new one for a different project.   It seems to me that this will cause subtle problems.  E.g., what if the original project stops?  In the store you'll think it is still running, since you just deleted the Table that would tell the store that the project stopped.

It seems to me that was is needed is a way to **actually** add a single project by having a way to make a changefeed to track what happens with a given list of projects.  That doesn't exist yet, and you didn't write it, and it's maybe not easy.  But I don't see how you can possibly solve this problem without doing that... unless you explicitly and purposely choose to make the information in the projects table be wrong.  Maybe that is OK for some use case.

Reply:
Well, unfortunately I don't know anything about this. No, my goal is not to make the table wrong. My only goal was to be able to switch between projects.

Reply:
Can you rename the function then?   E.g., shouldn't it be called "switch_between_projects"?

Reply:
yes, `switch_to_project` sounds good to me